### DISCUSSION TITLE
Using custom backend endpoint instead of localhost

### DISCUSSION THREAD

**Discussion by andybuki** [jbaiter/pdiiif#11] [Q&A] [answered]
Hi Johannes, I'm not familiar with typescript, that's why may be I wrong. 
I took a look at your repository, where you call http://localhost:31337/api.
Is it better to call the server instead of a localhost?

**Comment by jbaiter** [comment, marked as answer]
I assume you're refering to this piece of code in `pdiiif-web/src/main.ts`?

https://github.com/jbaiter/pdiiif/blob/1ebc08e802ffa29dda0d563f40dd38f5f7ad123d/pdiiif-web/src/main.ts#L48-L52

This is just a default value for the address where the backend API is located (for generating the cover page and for the server-based fallback). You can simply override this by setting the `PDIIIF_API_ENDPOINT` environment variable during the build (`import.meta.env` is resolved at build time, [see the vite docs here](https://vitejs.dev/guide/env-and-mode.html)).

For example, to use `https://example.com/pdiiif/api` as the backend location:

```
cd pdiiif-web
PDIIIF_API_ENDPOINT=https://example.com/pdiiif/api pnpm run build
```

The resulting build in `dist` will then contact that address for cover pages and fallbacks.

The default values if the environment variable is not set is as follows:

- Development builds (i.e. with `pnpm run dev`) use `http://localhost:31337/api`, assuming `pdiiif-api` is running there
- Production builds (i.e. with `pnpm run build`) use the location where the app is hosted and append `/api`.

**Reply by andybuki** [reply]
Thanks a lot for the explanation.
